client/src/app/editor/file-outputs/file-outputs-table.component.html

```diff
@@ -0,0 +1,58 @@
+<mat-table [@staggerIn]="outputFiles" [dataSource]="outputFiles">
+  <ng-container matColumnDef="name">
+    <mat-header-cell *matHeaderCellDef>Name</mat-header-cell>
+    <mat-cell *matCellDef="let row">
+      <div class="text-ellipsis">
+        <mat-icon>description</mat-icon> {{row.name}}
+      </div>
+    </mat-cell>
+  </ng-container>
+
+  <ng-container matColumnDef="created">
+    <mat-header-cell *matHeaderCellDef>Created</mat-header-cell>
+    <mat-cell *matCellDef="let row"> {{row.created_at | date}} </mat-cell>
+  </ng-container>
+
+  <ng-container matColumnDef="size">
+    <mat-header-cell *matHeaderCellDef>Size</mat-header-cell>
+    <mat-cell *matCellDef="let row">{{row.size_bytes | digitalFormatUnit}}</mat-cell>
+  </ng-container>
+
+  <ng-container matColumnDef="contentType">
+    <mat-header-cell *matHeaderCellDef>Content Type</mat-header-cell>
+    <mat-cell *matCellDef="let row">{{row.content_type || '-'}}</mat-cell>
+  </ng-container>
+
+  <ng-container matColumnDef="actions">
+    <mat-header-cell *matHeaderCellDef></mat-header-cell>
+    <mat-cell *matCellDef="let row">
+      <button mat-icon-button color="primary" matTooltip="Preview" (click)="preview.emit(row)" [disabled]="!isImage(row.name)"
+        title="Preview file">
+        <mat-icon>visibility</mat-icon>
+      </button>
+
+      <button mat-icon-button color="primary" matTooltip="Copy API Link" ngxClipboard [cbContent]="getApiLink(row)" (cbOnSuccess)="copyDone.emit()"
+        (cbOnError)="copyDone(true)" title="Copy API link">
+        <mat-icon>link</mat-icon>
+      </button>
+
+      <a mat-icon-button color="primary" matTooltip="Download" [attr.download]="row.name" target="_blank" [href]="row.download_url"
+        title="Download file">
+        <mat-icon>file_download</mat-icon>
+      </a>
+    </mat-cell>
+  </ng-container>
+
+  <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
+  <mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row>
+</mat-table>
+<div class="message" *ngIf="isLoading && !isExecuting && !outputFiles?.length">
+  <mat-progress-spinner mode="indeterminate"></mat-progress-spinner>
+  Loading output files...
+</div>
+<div class="message" *ngIf="!isLoading && !isExecuting && !outputFiles?.length">
+  This execution has no output files.
+</div>
+<div class="message" *ngIf="!outputFiles?.length && isExecuting">
+  No output files yet, but hold tight, we are listening for them in realtime...
+</div>
```

Comment:
Those three blocks could be refactored into:

```
<div class="message" *ngIf="!outputFiles?.length">
  <ng-container *ngIf="...">
   ...
  </ng-container>
  <ng-container *ngIf="...">
   ...
   </ng-container>
   ...
</div>
```

Reply:
Done.